pkg/api/registry/registry.proto

```diff
@@ -22,61 +22,48 @@ message Destination {
     uint32 weight = 2;
 }
 
-message NetworkServiceManager {
-    string name = 1;
-    string url = 2;
-    google.protobuf.Timestamp expiration_time = 3;
-    string state = 4;
+message NetworkServiceLabels {
+    map<string, string> labels = 1;
 }
 
 message NetworkServiceEndpoint {
     string name = 1;
-    string payload = 2;
-    string network_service_name = 3;
-    string network_service_manager_name = 4;
-    map<string, string> labels = 5;
-    string state = 6;
-
+    repeated string network_service_name = 2;
+    map<string, NetworkServiceLabels> labels = 3;
     // Endpoint callback address, could be any of unix/tcp/callback schema ones.
-    string url = 7;
+    string url = 4;
+    // Network service manager URL for the endpoint.
+    string network_service_manger_url = 5;
+    google.protobuf.Timestamp expiration_time = 6;
 }
 
-message FindNetworkServiceRequest {
-    string network_service_name = 1;
+service NetworkServiceEndpointRegistry {
+    rpc RegisterOrUpdate (NetworkServiceEndpoint) returns (NetworkServiceEndpoint);
+    rpc Monitor (NetworkServiceEndpointQuery) returns (stream NetworkServiceEndpoint);
+    rpc Find (NetworkServiceEndpointQuery) returns (stream NetworkServiceEndpointList);
+    rpc Unregister (NetworkServiceEndpoint) returns (google.protobuf.Empty);
 }
 
-message FindNetworkServiceResponse {
-    string payload = 1;
-    NetworkService network_service = 2;
-    map<string, NetworkServiceManager> network_service_managers = 3;
-    repeated NetworkServiceEndpoint network_service_endpoints = 4;
+service NetworkServiceRegistry {
+    rpc RegisterOrUpdate (NetworkService) returns (NetworkService);
```

Comment:
I think RegisterOrUpdate is same as Register for this context, so why we need so long function name :-)

Reply:
ok, fixed.